src/components/search/SearchTypesDirectives.js

```diff
@@ -443,8 +472,17 @@ goog.require('ga_urlutils_service');
                   features: [f],
                   onCloseCB: angular.noop
                 });
-                gaPreviewFeatures.zoom($scope.map,
-                    geojsonParser.readFeature(f));
+                var feature = geojsonParser.readFeature(f);
+                var ol3d = $scope.ol3d;
+                if (ol3d && ol3d.getEnabled()) {
+                  var extent = feature.getGeometry().getExtent();
+                  extent = gaPreviewFeatures.getMinimalExtent(extent);
+                  ol3d.getCesiumScene().camera.flyTo({
+                    destination: extentToRectangle(extent)
+                  });
```

Comment:
Why not putting this in the `gaPreviewFeatures.zoom`  adding ol3d as parameter of the function. We use this function in other place,  I guess.